pkg/envoy/accesslog_server.go

```diff
@@ -164,16 +162,26 @@ func logRecord(endpointInfoRegistry logger.EndpointInfoRegistry, localEndpoint l
 		})
 	}
 
-	r := logger.NewLogRecord(endpointInfoRegistry, localEndpoint, GetFlowType(pblog), pblog.IsIngress,
+	flowType := GetFlowType(pblog)
+	// Our accesslogs from Envoy do not flip the addressing for the response logs,
+	// swap them here so that the logs have the correct info.
+	// TODO (jrajahalme): Consider doing this at our Envoy filters instead?
+	var addrInfo logger.AddressingInfo
+	if flowType == accesslog.TypeResponse {
```

Comment:
This warrants a comment saying how the direction is interpreted (similar to above).

Reply:
I changed the comment to this to make this clearer:
```
	// Response access logs from Envoy inherit the source/destination info from the request log
	// message. Swap source/destination info here for the response logs so that they are
	// correct.
	// TODO (jrajahalme): Consider doing this at our Envoy filters instead?
```